src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -262,7 +325,7 @@ public PrivilegesEvaluatorResponse evaluate(PrivilegesEvaluationContext context)
             action0 = PutMappingAction.NAME;
         }
 
-        final PrivilegesEvaluatorResponse presponse = new PrivilegesEvaluatorResponse();
+        PrivilegesEvaluatorResponse presponse = new PrivilegesEvaluatorResponse();
```

Comment:
It looks like this presponse object is primarily used for exiting early from privileges evaluation, like in the case of snapshot, system index, protected index and PIT privilege. Since its using one of the constructors directly, the error message for privilege evaluation will not include the missing privileges. Do you think it makes sense to return the missing privileges for such cases?

i.e.

```
if (snapshotRestoreEvaluator.evaluate(request, task, action0, clusterInfoHolder, presponse).isComplete()) {
    if (!presponse.isAllowed()) {
        return PrivilegesEvaluatorResponse.insufficient(action0, context);
    } else {
        return presponse;
    }
}
```

Reply:
Looking at SnapshotRestoreEvaluator, there are actually quite a few different error scenarios:

- The operation is not allowed at all for normal users
- The operation includes protected indices

The actual presence of the permission is actually not checked in the SnapshotRestoreEvaluator, this is checked below in the "normal" privilege evaluation support. 

Thus, it  would be misleading to report the restore action as missing privilege, as adding that privilege won't fix the error. 

However, one could extend  SnapshotRestoreEvaluator to include the reason of the error in the `reason` attribute of presponse. That is however IMHO outside of the scope of this PR.